segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3Storage.java

```diff
@@ -263,6 +264,12 @@ private boolean syncExists(String streamSegmentName) {
     private SegmentProperties syncCreate(String streamSegmentName) throws StreamSegmentExistsException {
         long traceId = LoggerHelpers.traceEnter(log, "create", streamSegmentName);
 
+        /* To ensure put-if-absent semantics we need to use If-None-Match tag as described here:
```

Comment:
We need to improve this comment, here is what I'm missing:

- An example of what sort of problem we can have in the case we don't use put-if-absent. Just put down the example we have discussed offline:
```
Host2 is the current leader, while Host 1 is an old leader. Host 1 does not know that it lost leadership.
Host2 executes Create, Write1, Write2.
Host 1 executes Create, Write1
Host2 executes Write3 and fails.
```

- A link to the github issue for the object lib we are using that will fix `ifNoneMatch` instead of simply saying that it doesn't work. If there isn't such an issue, we can create it ourselves against their repo and reference it.
- A link to Pravega issue reporting the problem and saying that we need to fix it on our end once the object lib is fixed. I made a similar comment below, so it is not necessary to have in both places, pick one.